libmachine/provision/ubuntu.go

```diff
@@ -124,7 +124,7 @@ func (provisioner *UbuntuProvisioner) Hostname() (string, error) {
 
 func (provisioner *UbuntuProvisioner) SetHostname(hostname string) error {
 	cmd, err := provisioner.SSHCommand(fmt.Sprintf(
-		"sudo hostname %s && echo %q | sudo tee /etc/hostname && echo \"127.0.0.1 %s\" | sudo tee -a /etc/hosts",
+		"sudo hostname %s && echo %q | sudo tee /etc/hostname && sudo sed -i 's/^127.0.1.1.*/127.0.1.1 %s/g' /etc/hosts",
```

Comment:
Because we want to keep the 127.0.0.1 entry, ubuntu/debian use 127.0.1.1 for non "localhost" loopback hostnames:

https://www.debian.org/doc/manuals/debian-reference/ch05.en.html#_the_hostname_resolution